### PR TITLE
Make writes atomic, add unused "allowDowngrades" option to apt, fix a few recipe issues

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: policies/recipes/steps.go [hunk 1]
```diff
@@ -494,10 +505,10 @@ func writeScript(path, contents string) error {
 		f.Close()
 		return err
 	}
-	if err := f.Close(); err != nil {
+	if err := f.Chmod(0755); err != nil {
 		return err
 	}
-	if err := os.Chmod(path, 0755); err != nil {
+	if err := f.Close(); err != nil {
```
Comment (ryanwe) [inline policies/recipes/steps.go:511]: Should this use AtomicWrite too?

File: policies/recipes/steps.go [hunk 2]
```diff
@@ -80,18 +83,26 @@ func stepCopyFile(step *agentendpointpb.SoftwareRecipe_Step_CopyFile, artifacts
...
 	}
 
-	return nil
+	if err := tmp.Chmod(permissions); err != nil {
+		return err
+	}
+
+	if err := tmp.Close(); err != nil {
+		return err
+	}
```
Comment (ryanwe) [inline policies/recipes/steps.go:103]: Would an AtomicFileCopy be helpful? The only difference is io.Copy vs. io.Write.

Comment (adjackura) [reply policies/recipes/steps.go:103]: Not needed by any other function right now, will add if I find other uses

**Review by ryanwe — APPROVED** [review]
/lgtm

**Pull request merged**